Allow multiple hosts to be defined when using ingress

guillaumelfv opened the pull request:
##### SUMMARY
Allow multiple hosts to be defined when using ingress.
Replace `hostname` and `ingress_tls_secret` with a dictionary to define ingress:
```yaml
ingress_hosts:
  - hostname: awx-demo.example.com
    tls_secret: example-com-tls
  - hostname: awx-sample.example.io
```
`ingress_hosts` default to empty and when defined `hostname` is mandatory but `tls_secret` is optional.

For the example above the ingress resources will be templatize as follow:
```yaml
  rules:
  - host: awx-demo.example.com
    http:
      paths:
      - backend:
          service:
            name: awx-service
            port:
              number: 80
        path: /
        pathType: Prefix
  - host: awx-sample.example.io
    http:
      paths:
      - backend:
          service:
            name: awx-service
            port:
              number: 80
        path: /
        pathType: Prefix
  tls:
  - hosts:
    - awx-demo.example.com
    secretName: example-com-tls
```

fixes #897 #1318 

##### ISSUE TYPE
 - New or Enhanced Feature

##### ADDITIONAL INFORMATION
`hostname` is removed from https://github.com/ansible/awx-operator#ingress-type 
```
hostname | Define the FQDN | {{ meta.name }}.example.com
```
and replaced with `ingress_hosts`
```
ingress_hosts | Define one or multiple FQDN with optional Secret that contains the TLS information | Empty string
```
This change is not backward compatible and anyone using ingress `hostname` spec will need to change to the new `ingress_hosts` spec.

Commit:
add new specs ingress_hosts

Commit:
fix new k8s_info api_version for deployment kind

Commit:
Replace hostname and ingress_tls_secret specs with ingress_hosts spec

djyasin committed:
Updating awx_test.yml to respect both web and task deployments.
- Extracts label information from both web and task pods
- Asserts web and task pod labels are correct respectively

thedoubl3j committed:
revert project persistance so that manual projects have a default path

move chmod chgrp for projects_persistence to different init

default init container is the awx-ee because and unable to run command chgrp

moving it into a different init container

note this is not needed for openshift so this is conditional on is_k8s

Commit:
Merge branch 'devel' into multiple_ingress_hosts

TheRealHaoLiu committed:
Add CA cert volume mount to rsyslog sidecar

fixes https://github.com/ansible/awx-operator/issues/1332

Commit:
manage new ingress_host tls secret backup separately

Commit:
Merge branch 'devel' into multiple_ingress_hosts

guillaumelfv committed:
Fix readme unexpected changes

TheRealHaoLiu committed:
Update awx_test.yml

- rename awx_pod to awx_web_pod to better reflect intention
- update `awx_version` block to also validate awx_task_pod
- add get awx_task_pod in additional label check block (fix CI in awx repo which does not run the validate awx_version block)
- add assertion for awx_task_pod extra label

guillaumelfv commented:
Whichever PR is merge first between this one and #1320 from will need to rebase and adapt the `ingress.yaml.j2`

thedoubl3j committed:
remove to_nice_yaml from tolerations in the task deployment which caused in malformed yaml to be rendered for the task deployment

djyasin committed:
Add ability to override annotations independently

- add web_annotations to override annotations for web pods
- add task_annotations to override annotations for task pods

TheRealHaoLiu committed:
Fix helm release problem (attempt 3) (#1350)

fix helm release result in helm index containing incorrect download link

Co-authored-by: Christian Adams <rooftopcellist@gmail.com>
Co-authored-by: John Westcott IV <32551173+john-westcott-iv@users.noreply.github.com>

rooftopcellist commented:
We will need to make this backwards compatible before we can merge.  For example, we can't remove `ingress_tls_secret` from the CRD (same with hostname).  And task logic needs to be added to enable `hostname` or `ingress_hosts`.

Also, the old CRD parameters need to be marked as Deprecated in the CRD description.

TheRealHaoLiu committed:
Fix lint error (#1352)

Check warning on line 44 in roles/installer/defaults/main.yml
44:2 [comments] missing starting space in comment

rooftopcellist committed:
Set securityContext.capabilities for kube-rbac-proxy container (#1354)

fosterseth committed:
fix development mode

tanganellilore committed:
fix chart package path on hel-release playbook

sdx23 committed:
Update README.md

Adaptations to use kustomize implicitly via kubectl apply -k

john-westcott-iv committed:
Changing helm-release again

rooftopcellist committed:
Fix linter warnings on PRs (#1372)

lumarel committed:
The ingress resulted in a double slash, if there is are slashes at the end of the ingress_path variable (#1374)

Co-authored-by: lumarel <lumarel@users.noreply.github.com>

rooftopcellist committed:
Add docs for how to run molecule tests with kind (#1375)

- test-local docker scenario was removed in https://github.com/ansible/awx-operator/commit/9a54ae293741cff695579834c86f3b4d17c43d36

fosterseth committed:
bump kustomize v4.5.7 to v5.0.1 (#1357)

Commit:
resolve conflict

guillaumelfv commented:
sorry @rooftopcellist i had to close this PR and open a new one #1377
